subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/internal/component/SelectionFailureHandlerIntegrationTest.groovy

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.internal.component
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.test.fixtures.dsl.GradleDsl
+
+import static org.gradle.test.fixtures.dsl.GradleDsl.*
+
+/**
+ * These tests demonstrate the behavior of the [SelectionFailureHandler] when a project has various
+ * variant selection failures.
+ */
+class SelectionFailureHandlerIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
Did you intend to include these changes too?

Reply:
Yes, they're very basic now, but I'll be expanding them later.  Not a super-cohesive PR unfortunately.